tools/wpt/run.py

```diff
@@ -487,6 +492,23 @@ def setup_wptrunner(venv, prompt=True, install_browser=False, **kwargs):
     setup_cls = product_setup[kwargs["product"]](venv, prompt, sub_product)
     setup_cls.install_requirements()
 
+    affected_revish = kwargs.pop("affected", None)
+    if affected_revish is not None:
+        files_changed, _ = testfiles.files_changed(
+            affected_revish, include_uncommitted=True, include_new=True)
+        # TODO(Hexcles): Perhaps use wptrunner.testloader.ManifestLoader here
+        # so that we can reuse it later in wptrunner instead of loading it
+        # again and remove the manifest-related code from testfiles.
+        tests_changed, tests_affected = testfiles.affected_testfiles(
+            files_changed, manifest_path=kwargs.get("manifest_path"), manifest_update=kwargs["manifest_update"])
+        test_list = tests_changed | tests_affected
+        logger.info("Identified %s affected tests" % len(test_list))
+        if not test_list:
```

Comment:
```suggestion
        if not test_list and not kwargs["test_list"]:
```

Or some other fix that still runs any explicitly listed tests.